### PR TITLE
Initial steps towards compatibility with wxWidgets-3.0

### PR DESCRIPTION
**Pull request by leio**
These are some initial steps towards being able to build against stable wxWidgets-3.0.
The parts of #300 that I feel are ready for review.

This is not marked as "compiled on my machine", because I don't have wxWidgets-3.1. With the extra WIP changes found in #300, it compiles fine against my distro wxGTK-3.0.5.1 on my machine.

<details>
<summary>Checklist</summary>

- [x] I have signed off my commits using `-s` or `Signed-off-by`\* (See: [Contributing § DCO](https://github.com/tenacityteam/tenacity/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))
- [ ] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code\*
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving\*
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"\*

\* indicates required

</details>

### CODE REVIEW

File: src/widgets/Meter.cpp [hunk 1]
```diff
@@ -471,6 +471,15 @@ void MeterPanel::OnErase(wxEraseEvent & WXUNUSED(event))
    // Ignore it to prevent flashing
 }
 
+static double GetColourLuminance(wxColour color)
+{
+#if wxCHECK_VERSION(3, 1, 0)
+    return color.GetLuminance();
+#else
+    return (0.299*color.Red() + 0.587*color.Green() + 0.114*color.Blue()) / 255.0;
```
Comment (Be-ing) [inline src/widgets/Meter.cpp:480]: Is this just copying the implementation from wxWidgets 3.1? Where do these magic numbers come from?

File: libraries/lib-string-utils/CodeConversions.cpp [hunk 2]
```diff
@@ -28,7 +28,16 @@ std::string ToUTF8 (const wchar_t* wstr)
 
 std::string ToUTF8 (const wxString& wstr)
 {
+#if wxCHECK_VERSION(3, 1, 1)
     return wstr.ToStdString (wxGet_wxConvUTF8 ());
+#else
+#if !wxUSE_UNICODE
```
Comment (Be-ing) [inline libraries/lib-string-utils/CodeConversions.cpp]: I'd prefer a single line `#elseif` than creating nested `if` preprocessor blocks

File: src/widgets/Meter.cpp (hunk 1 above)
Comment (leio) [reply src/widgets/Meter.cpp:480]: These are standard suggested values for such conversion. And yes, wxWidgets happens to also simply use that.

https://www.opengl.org/archives/resources/code/samples/advanced/advanced97/notes/node140.html
https://stackoverflow.com/questions/596216/formula-to-determine-perceived-brightness-of-rgb-color

and so on.

File: src/widgets/FileConfig.cpp [hunk 3]
```diff
@@ -63,7 +63,10 @@ void FileConfig::Init()
       // because we don't use the wxFileConfig::Flush() method and so the wxFileConfig dirty
       // flag never gets reset. During deletion, the dirty flag is checked and a Flush()
       // performed. This can (and probably will) create bogus temporary files.
```
Comment (Be-ing) [inline src/widgets/FileConfig.cpp:65]: I agree that "bogus temporary files" are a trivial issue in comparison to breaking wxWidgets 3.0 compatibility.

File: libraries/lib-string-utils/CodeConversions.cpp (hunk 2 above)
Comment (leio) [reply libraries/lib-string-utils/CodeConversions.cpp]: So one branch is if we've got 3.1, the other branch is if we don't.

By not nesting it, it reads conceptually wrong to me. But given the limitations on preprocessor #if styling, maybe it makes sense to change, yes.

File: src/widgets/Meter.cpp (hunk 1 above)
Comment (Be-ing) [reply src/widgets/Meter.cpp:480]: Please link to one of these in a code comment.

File: libraries/lib-string-utils/CodeConversions.cpp (hunk 2 above)
Comment (leio) [reply libraries/lib-string-utils/CodeConversions.cpp]: Change this.

File: src/widgets/Meter.cpp (hunk 1 above)
Comment (leio) [reply src/widgets/Meter.cpp:480]: I do not agree with putting random links into the codebase to what is a bog standard luminosity calculation math, used pretty much everywhere for luminosity calculation out of RGB. I've added a comment explaining this instead on top of it.

Comment (Be-ing) [reply src/widgets/Meter.cpp:480]: Okay, so long as there is some explanation of what these numeric constants are.

File: libraries/lib-string-utils/CodeConversions.cpp (hunk 2 above)
Comment (Be-ing) [reply libraries/lib-string-utils/CodeConversions.cpp]: Oops, it's `#elif`, not `#elseif`

Comment (leio) [reply libraries/lib-string-utils/CodeConversions.cpp]: My gcc10 was happy with #elseif too for some reason (and does fail if I do stuff like `#else if`, so it is parsing that line), so didn't even think twice about it :)

File: libraries/lib-strings/TranslatableString.cpp [hunk 4]
```diff
@@ -97,7 +97,11 @@ wxString TranslatableString::DoSubstitute( const Formatter &formatter,
    return formatter
       ? formatter( format, debug ? Request::DebugFormat : Request::Format )
       : // come here for most translatable strings, which have no formatting
+#if !wxCHECK_VERSION(3, 1, 1)
+         ( debug ? format : wxGetTranslation( context.IsEmpty() ? format : wxString(context) + wxString('\x04') + format, wxString{} ) );
```
Comment (leio) [inline libraries/lib-strings/TranslatableString.cpp]: I found a place that actually exercises this code and it isn't quite working right yet (at runtime, with wx3.0; all of course as good as before with 3.1). Will need to see what's going on here before merging this commit.

If everything else is fine, I could also drop this commit and we can get the rest in, there'll be follow-ups anyhow (crash dialog, XDG paths and cmake updates)

Comment (Be-ing) [reply libraries/lib-strings/TranslatableString.cpp]: Yes, let's merge what's working already and you can cherry pick the WIP to another branch.

**Review by Be-ing — APPROVED** [review]

File: libraries/lib-strings/TranslatableString.cpp (hunk 4 above)
Comment (leio) [reply libraries/lib-strings/TranslatableString.cpp]: I had dropped this commit here a while back

**Pull request merged**